### ISSUE TITLE
Embeddable ignores `Unique` decorator when running migration:create

### ISSUE THREAD

**Issue by Tronikelis** [mikro-orm/mikro-orm#3879]
**Describe the bug**
When running `npx mikro-orm migration:create` `@Unique` decorator gets ignored, but `@Property({ unique: true })` does not. When using embeddables

**To Reproduce**
I have created a repo https://github.com/Tronikelis/mikro-orm-issue-2

1. Create an entity with an embeddable
2. The embeddable should have a unique property `@Unique`
3. `migration:create` => does not create a unique constraint
4. Change the embeddable property to `@Property({ unique: true })` => now it creates is

With `@Unique`:

```typescript
import { Migration } from '@mikro-orm/migrations';

export class Migration20221228181129 extends Migration {

  async up(): Promise<void> {
    this.addSql('create table "user" ("id" serial primary key, "metadata_slug" text not null);');
  }

  async down(): Promise<void> {
    this.addSql('drop table if exists "user" cascade;');
  }

}
```

With `@Property({ unique: true })`:

```typescript
import { Migration } from '@mikro-orm/migrations';

export class Migration20221228181216 extends Migration {

  async up(): Promise<void> {
    this.addSql('create table "user" ("id" serial primary key, "metadata_slug" text not null);');
    this.addSql('alter table "user" add constraint "user_metadata_slug_unique" unique ("metadata_slug");');
  }

  async down(): Promise<void> {
    this.addSql('drop table if exists "user" cascade;');
  }

}
```

**Expected behavior**
Both of the methods should work the same

**Versions**

| Dependency | Version |
| - | - |
| node | 16.19.0 |
| typescript | 4.9.4 |
| mikro-orm | 5.6.2 |
| your-driver | postgres |

**Comment by B4nan** [comment]
Not just unique constraints, same applies for regular indexes. I believe we have at least 2 other issues about this, the main reason is the embeddables can be used in object mode, so this feature implies support for indexes on JSON properties. I'd like to revisit this in v6.

**Comment by B4nan** [comment]
Closing as implemented in v6 via #4736
